packages/number-field/src/NumberField.ts

```diff
@@ -495,7 +495,17 @@ export class NumberField extends TextfieldBase {
         // Step shouldn't validate when 0...
         if (this.step) {
             const min = typeof this.min !== 'undefined' ? this.min : 0;
-            const moduloStep = (value - min) % this.step;
+            const digitsAfterDecimals =
+                this.step != Math.floor(this.step)
+                    ? this.step.toString().split('.')[1].length
+                    : 0;
+            const formatter = new Intl.NumberFormat('en-US', {
+                maximumFractionDigits: digitsAfterDecimals,
+            });
+            const moduloStep = parseFloat(
+                formatter.format((value - min) % this.step)
```

Comment:
How will you catch if the value is less than min?

Reply:
Cecking if value is less than `min` or more than `max` is handled in `valueWithLimits` function, which is being called on line 491 at the beginning of `validateInput`